crates/notion-core/src/manifest/serial.rs

```diff
@@ -76,6 +81,31 @@ pub struct Engines {
     pub node: String,
 }
 
+impl Engines {
+    /// Handle deserialization where users may have supplied bad values for the
+    /// `node` key in a `package.json`.
+    ///
+    /// We are intentionally extremely permissive: we simply return `None` for
+    /// all scenarios other than finding a valid POJO like `{ node: <spec> }`
+    /// because we are happy to use that information if it is available, but if
+    /// it is either unavailable or malformed, we simply fall back to our normal
+    /// handling.
+    pub fn deserialize<'de, D>(d: D) -> Result<Option<Engines>, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        Deserialize::deserialize(d).map(|value: Option<Value>| match value {
+            Some(Value::Object(object)) => match object.get("node") {
```

Comment:
We can probably simplify the flow of this block slightly using `if let`, since the fail case is the same in both matches:

```rust
Deserialize::deserialize(d).map(|value: Option<Value>| {
    if let Some(Value::Object(object)) = value {
        if let Some(Value::String(ref node)) = object.get("node") {
            return Some(Engines {
                node: node.to_string(),
            });
        }
    }
    None
})
```

I leave it up to you if you think that's better or not, it seems pretty close to me.

Reply:
Totally fair to raise – I went back and forth on this one myself. I ultimately went with the `match` statements for two (totally just personal preference) reasons:

1. It's both shorter (by two lines) and less deeply nested (by one level of indentation) to use the `match` expressions.
2. I lean toward expression evaluations, rather than control flow with early `return`s—for whatever reason they're just clearer to me.

Neither of those is particularly binding, but for those two reasons my inclination would be to leave it as is. If you or others find it clearer with the `if let's, I'm totally willing to change it!

Reply:
Fair enough :+1: